### PR TITLE
EPMRPP-33143 - Infoline.EPMRPP-33162 - defect type tooltips. Project controller.

### PR DESCRIPTION
**Pull request by ArtsiomYeliseyenka**
…controller

### CODE REVIEW

File: app/src/components/main/tooltips/hoverableTooltip/hoverableTooltip.scss [hunk 1]
```diff
@@ -40,6 +41,22 @@
     border-left: 6px solid transparent;
     border-right: 6px solid transparent;
   }
+
+  &.noArrow {
```
Comment (DiscoElevator) [inline app/src/components/main/tooltips/hoverableTooltip/hoverableTooltip.scss]: no-arrow

File: app/src/components/main/tooltips/hoverableTooltip/hoverableTooltip.scss [hunk 2]
```diff
@@ -40,6 +41,22 @@
...
+    &::before, &::after  {
+      display: none;
+    }
+  }
+  &.noMobile {
```
Comment (DiscoElevator) [inline app/src/components/main/tooltips/hoverableTooltip/hoverableTooltip.scss]: no-mobile

File: app/src/components/main/tooltips/hoverableTooltip/hoverableTooltip.scss [hunk 3]
```diff
@@ -40,6 +41,22 @@
...
+    @media (max-width: $SCREEN_XS_MAX) {
+      display: none;
+    }
+  }
+  &.desktopOnly {
```
Comment (DiscoElevator) [inline app/src/components/main/tooltips/hoverableTooltip/hoverableTooltip.scss]: desktop-only

File: app/src/components/main/tooltips/hoverableTooltip/withHoverableTooltip.jsx [hunk 4]
```diff
@@ -16,16 +16,38 @@ export const withHoverableTooltip = ({ TooltipComponent, data }) => WrappedCompo
...
     render() {
       return (
         <div className={cx('hover-rect')} ref={(hoverRect) => { this.hoverRect = hoverRect; }} >
           <WrappedComponent {...this.props}>
             { this.props.children }
           </WrappedComponent>
-          <div ref={(tooltip) => { this.tooltip = tooltip; }} className={cx('hoverable-tooltip')}>
+          <div
+            ref={(tooltip) => { this.tooltip = tooltip; }}
+            className={cx({
```
Comment (DiscoElevator) [inline app/src/components/main/tooltips/hoverableTooltip/withHoverableTooltip.jsx]: `className={cx('hoverable-tooltip', {`

File: app/src/components/main/tooltips/hoverableTooltip/withHoverableTooltip.jsx [hunk 5]
```diff
@@ -16,16 +16,38 @@ export const withHoverableTooltip = ({ TooltipComponent, data }) => WrappedCompo
     };
     componentDidMount() {
       const tooltipWidth = data.width || DEFAULT_TOOLTIP_WIDTH;
+      let left = 0;
       this.tooltip.style.width = `${tooltipWidth}px`;
-      this.tooltip.style.left = `${(this.hoverRect.offsetWidth / 2) - (tooltipWidth / 2)}px`;
+      switch (data.align) {
+        case 'left':
```
Comment (DiscoElevator) [inline app/src/components/main/tooltips/hoverableTooltip/withHoverableTooltip.jsx]: 'left' and 'right' should be constants for both tooltip types

File: app/src/components/main/tooltips/tooltip/tooltip.scss [hunk 6]
```diff
@@ -24,6 +24,12 @@
     border-left: 6px solid transparent;
     border-right: 6px solid transparent;
   }
+
+  &.noArrow {
```
Comment (DiscoElevator) [inline app/src/components/main/tooltips/tooltip/tooltip.scss]: no-arrow

File: app/src/controllers/project/reducer.js [hunk 7]
```diff
@@ -0,0 +1,10 @@
+import { INITIAL_STATE, FETCH_PROJECT_SUCCESS } from './constants';
+
+export const projectReducer = (state = INITIAL_STATE, { type, payload }) => {
+  switch (type) {
+    case FETCH_PROJECT_SUCCESS:
+      return Object.assign({}, state, payload);
```
Comment (DiscoElevator) [inline app/src/controllers/project/reducer.js]: `{ ...state, ...payload }`

File: app/src/pages/inside/launchesPage/infoLine/infoLine.jsx [hunk 8]
```diff
@@ -0,0 +1,76 @@
...
+        <div className={cx('duration')}>
+          <FormattedMessage id="InfoLine.duration" defaultMessage="Duration" />
+          <div className={cx('duration-value')}>
+            {/* TODO change on duration calculation */}
+            <Duration duration={'7s'} />
+          </div>
+        </div>
+        <div className={cx('defect-types')}>
+          {
+            Object.keys(defects).map(
```
Comment (DiscoElevator) [inline app/src/pages/inside/launchesPage/infoLine/infoLine.jsx]: `Object.keys(defects).map(key => (`

File: app/src/pages/inside/launchesPage/defectTypeTooltip/defectTypeTooltip.jsx [hunk 9]
```diff
@@ -0,0 +1,62 @@
...
+    return (
+      <div className={cx('defect-type-tooltip')}>
+        {
+          (this.props.type !== 'to_investigate') &&
+          <div className={cx('total-item')}>
+            <div className={cx('name')}>
+              <div className={cx('circle')} style={{ backgroundColor: defectConfig[0].color }} />
+              { formatMessage(messages[`${this.props.type}_total`]) }
+            </div>
+            <a href="/" className={cx('value')}>
```
Comment (DiscoElevator) [inline app/src/pages/inside/launchesPage/defectTypeTooltip/defectTypeTooltip.jsx]: Whole row should be clickable

File: app/src/pages/inside/launchesPage/defectTypeTooltip/defectTypeTooltip.jsx [hunk 10]
```diff
@@ -0,0 +1,62 @@
...
+          </div>
+        }
+        {
+          defectConfig.map(item => (
+            <div key={item.locator} className={cx('item')}>
+              <div className={cx('name')}>
+                <div className={cx('circle')} style={{ backgroundColor: item.color }} />
+                { item.longName }
+              </div>
+              <a href="/" className={cx('value')}>
```
Comment (DiscoElevator) [inline app/src/pages/inside/launchesPage/defectTypeTooltip/defectTypeTooltip.jsx]: Whole row should be clickable

File: app/src/components/main/tooltips/hoverableTooltip/hoverableTooltip.scss (hunk 1 above)
Comment (ArtsiomYeliseyenka) [reply app/src/components/main/tooltips/hoverableTooltip/hoverableTooltip.scss]: fixed

File: app/src/components/main/tooltips/hoverableTooltip/hoverableTooltip.scss (hunk 2 above)
Comment (ArtsiomYeliseyenka) [reply app/src/components/main/tooltips/hoverableTooltip/hoverableTooltip.scss]: fixed

File: app/src/components/main/tooltips/hoverableTooltip/hoverableTooltip.scss (hunk 3 above)
Comment (ArtsiomYeliseyenka) [reply app/src/components/main/tooltips/hoverableTooltip/hoverableTooltip.scss]: fixed

File: app/src/components/main/tooltips/hoverableTooltip/withHoverableTooltip.jsx (hunk 4 above)
Comment (ArtsiomYeliseyenka) [reply app/src/components/main/tooltips/hoverableTooltip/withHoverableTooltip.jsx]: fixed

File: app/src/components/main/tooltips/tooltip/tooltip.scss (hunk 6 above)
Comment (ArtsiomYeliseyenka) [reply app/src/components/main/tooltips/tooltip/tooltip.scss]: fixed

File: app/src/controllers/project/reducer.js (hunk 7 above)
Comment (ArtsiomYeliseyenka) [reply app/src/controllers/project/reducer.js]: fixed

File: app/src/pages/inside/launchesPage/infoLine/infoLine.jsx (hunk 8 above)
Comment (ArtsiomYeliseyenka) [reply app/src/pages/inside/launchesPage/infoLine/infoLine.jsx]: fixed

File: app/src/components/main/tooltips/hoverableTooltip/withHoverableTooltip.jsx (hunk 5 above)
Comment (ArtsiomYeliseyenka) [reply app/src/components/main/tooltips/hoverableTooltip/withHoverableTooltip.jsx]: fixed

File: app/src/pages/inside/launchesPage/defectTypeTooltip/defectTypeTooltip.jsx (hunk 9 above)
Comment (ArtsiomYeliseyenka) [reply app/src/pages/inside/launchesPage/defectTypeTooltip/defectTypeTooltip.jsx]: fixed

File: app/src/pages/inside/launchesPage/defectTypeTooltip/defectTypeTooltip.jsx (hunk 10 above)
Comment (ArtsiomYeliseyenka) [reply app/src/pages/inside/launchesPage/defectTypeTooltip/defectTypeTooltip.jsx]: fixed

**Pull request merged**